src/js/angular/core/directives/queryeditor/query-editor.directive.js

```diff
@@ -574,21 +586,66 @@ function queryEditorDirective($timeout, $location, toastr, $repositories, Sparql
             window.yasr = yasr;
             yasr.afterCopy = afterCopy;
             yasr.getQueryResultsAsFormat = function (downloadFormat) {
-                // Simple cross-browser download with a form
-                const $wbDownload = $('#wb-download');
-                $wbDownload.attr('action', 'repositories/' + $repositories.getActiveRepository());
-                $('#wb-download-query').val(scope.currentQuery.query);
-                if (window.editor.getValue() !== scope.currentQuery.query) {
+                const exportFormatWithErrorHandling = scope.exportFormats.filter((format) => format.name.includes('JSON'));
```

Comment:
IMO will better if you use "type" or "extension" instead "name". Unlikely but the "name" can be changed .

Reply:
Refactored that